### PR TITLE
fix: serve csv files directly from s3

### LINKED ISSUE #300: Pipeline stops importing after a certaine number of rows
The behavior is reproducible and stable (import always stops on the same position)

Steps to reproduce (a)

- Map and import [CH_yearly_air_immission_basetable_so2_annualmean.zip](https://github.com/zazuko/cube-creator/files/5660283/CH_yearly_air_immission_basetable_so2_annualmean.zip) (all columns as observation table, use default mapping options)
  - only 4 rows are imported (row 5 has blank on column "value_remark")

Steps to reproduce (b)

- Map and import [CH_yearly_air_immission_basetable_so2_annualmean.zip](https://github.com/zazuko/cube-creator/files/5660283/CH_yearly_air_immission_basetable_so2_annualmean.zip) (all columns **except "value_remark"** as observation table, use default mapping options)
  - only 20 rows are displayed in preview (even with page size 50 or 100)
  - there is no structural difference between the last line imported (2003) and the first one not imported (2004)

Content of CSV: (a) imports 1984 to 1987, (b) 1984 to 2003
```
"station_id","pollutant_id","aggregation_id","limitvalue","year","value","unit_id","value_remark"
"blBAS","so2","annualmean",30,1984,31.9,"µg/m3","incomplete series"
"blBAS","so2","annualmean",30,1985,40.2,"µg/m3","incomplete series"
"blBAS","so2","annualmean",30,1986,33.6,"µg/m3","incomplete series"
"blBAS","so2","annualmean",30,1987,33,"µg/m3","incomplete series"
"blBAS","so2","annualmean",30,1988,17.4,"µg/m3",
"blBAS","so2","annualmean",30,1989,18.5,"µg/m3",
"blBAS","so2","annualmean",30,1990,14.1,"µg/m3",
"blBAS","so2","annualmean",30,1991,13.6,"µg/m3",
"blBAS","so2","annualmean",30,1992,11,"µg/m3",
"blBAS","so2","annualmean",30,1993,9.2,"µg/m3",
"blBAS","so2","annualmean",30,1994,6.7,"µg/m3",
"blBAS","so2","annualmean",30,1995,7.2,"µg/m3",
"blBAS","so2","annualmean",30,1996,7.2,"µg/m3",
"blBAS","so2","annualmean",30,1997,7.3,"µg/m3",
"blBAS","so2","annualmean",30,1998,7.1,"µg/m3",
"blBAS","so2","annualmean",30,1999,5.4,"µg/m3",
"blBAS","so2","annualmean",30,2000,4.7,"µg/m3",
"blBAS","so2","annualmean",30,2001,4.4,"µg/m3",
"blBAS","so2","annualmean",30,2002,6.1,"µg/m3",
"blBAS","so2","annualmean",30,2003,5.4,"µg/m3",
"blBAS","so2","annualmean",30,2004,3.7,"µg/m3",
"blBAS","so2","annualmean",30,2005,4.3,"µg/m3",
"blBAS","so2","annualmean",30,2006,4.4,"µg/m3",
"blBAS","so2","annualmean",30,2007,3.2,"µg/m3",
...
```

### PR DESCRIPTION
**Pull request by tpluscode**
fixes #300 

We originally intended the pipeline to only transform a subset of the data for the preview and it remained like that. Since that is not the case, I remove the "get head" and the csv is served directly from s3 using a resigned link.

I'm considering removing the `accept: text/csv` handler completely (now a redirect) given that the `schema:associatedMedia/schema:contentUrl` is in fact a functioning download link. @martinmaillard is there any reason to keep the csv handling logic? The only downside is that the link has an expiration date so that if you wait too long you'll need to fetch the source again

### CODE REVIEW

File: e2e-tests/csv-source/get-csv.hydra [hunk 1]
```diff
@@ -62,7 +62,7 @@ With Class cc:CSVSource {
             Accept "text/csv"
         } => {
             Expect Status 200
-            Expect Header Content-Type 'text/csv; charset=utf-8'
+            Expect Header Content-Type 'application/octet-stream'
```
Comment (tpluscode) [inline e2e-tests/csv-source/get-csv.hydra:65]: Minio serves like that

File: cli/lib/csv.ts [hunk 2]
```diff
@@ -6,21 +6,32 @@ import { Stream } from 'rdf-js'
...
+      csvStream.emit('error', new Error('Failed to load Source. Response was: ' + response?.xhr.statusText))
+      return
+    }
+
+    if (!representation.root.associatedMedia.contentUrl) {
+      csvStream.emit('error', new Error('Failed to load CSV. Missing link from source'))
+      return
+    }
+
+    const csvResponse = await fetch(representation.root.associatedMedia.contentUrl.value)
```
Comment (tpluscode) [inline cli/lib/csv.ts:27]: For some reason which I cannot understand minio fails to serve the file when fetched using alcaeus 🤷

**Review by lucafurrer — APPROVED** [review]
Looks good, have not tested it though

**Review by martinmaillard — APPROVED** [review]
The download link expiration is totally acceptable

**Pull request merged**